app/routes/dataurl.tsx

```diff
@@ -0,0 +1,48 @@
+import type { MetaFunction } from "@remix-run/cloudflare";
+import DataUrlDisplay from "~/components/dataurl-display";
+
+export const meta: MetaFunction = () => ({
+  title: "DataUrl | Utiliti",
+});
+
+function isImage(dataUrl: string, fileType: string): Promise<boolean> {
+  return new Promise((resolve) => {
+    const image = new Image();
+    image.onload = () => {
+      resolve(true);
+    };
+    image.onerror = () => {
+      resolve(false);
+    };
+    image.src = `data:${fileType};base64,${dataUrl}`;
+  });
+}
+
+async function display(text: string): Promise<string> {
+  try {
+    if (text.substring(0, 5) === "data:") {
```

Comment:
subjective: might be nicer to move this outside of the try and do an early return